From 1eb68f65fd73443d862ca5f36f3b6041e2b0f13a Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 10 Mar 2022 18:53:13 -0800
Subject: [PATCH] configure: Limit the header search to sysroot

Hardcoded paths in configure.ac means it starts to poke at host system
include paths, which is least we want in a cross build, therefore prefix
these paths with STAGING_SYSROOT

Upstream-Status: Inappropriate [OE Specific]

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 configure.ac | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -907,7 +907,7 @@ AC_DEFUN([PCP_CHECK_PYTHON_HEADER],
   [ PY_MAJOR="$1"
     PY_MINOR="$2"
     PY_VERSION="python$PY_MAJOR.$PY_MINOR"
-    PY_INCLUDE_DIRS="-I/usr/include/${PY_VERSION}m -I/usr/include/${PY_VERSION}"
+    PY_INCLUDE_DIRS="-I${STAGING_INCDIR}/${PY_VERSION}m -I${STAGING_INCDIR}/${PY_VERSION}"
     saved_CFLAGS="$CFLAGS"
     saved_CPPFLAGS="$CPPFLAGS"
     CFLAGS="$CFLAGS $PY_INCLUDE_DIRS"
@@ -1603,7 +1603,7 @@ AC_SUBST(qmake)
 
 enable_qt3d=false
 AS_IF([test "x$do_qt3d" != "xno"], [
-    if test -f /usr/include/Coin3/Inventor/Qt/SoQt.h -o -f /usr/include/Inventor/Qt/SoQt.h
+    if test -f ${STAGING_INCDIR}/Coin3/Inventor/Qt/SoQt.h -o -f ${STAGING_INCDIR}/Inventor/Qt/SoQt.h
     then
 	enable_qt3d=$enable_qt
     elif test x"$do_qt3d" = x"yes"
@@ -3699,7 +3699,7 @@ AC_SUBST(enable_decompression)
 AC_SUBST(enable_lzma)
 
 dnl check for array sessions
-if test -f /usr/include/sn/arsess.h
+if test -f ${STAGING_INCDIR}/sn/arsess.h
 then
     pcp_mpi_dirs=libpcp_mpi\ libpcp_mpiread
 else
